fix: Further updates to Nautobot api response logging

stevekeay opened the pull request:
Logs more information, but limits the size of any raw payloads we dump out to the logger.

Also extracts the "error" key instead of logging the whole json payload.

```
neutron_understack.nautobot.NautobotRequestError: Nautobot API ERROR 404 for https://nautobot.dev.undercloud.rackspace.net/testing-failures POST {'payload1': 123}: {'body': 'b\'\\n\\n\\n\\n<!DOCTYPE html>\\n<html lang="en">\\n<head>\\n    '
         '<title>Page Not Found - Nautobot</title>\\n    \\n\\n\\n\\n    <link '
         'rel="stylesheet" id="template-theme"\\n          '
         'href="/template.css">\\n    <lin...10971 more chars'}
```

cardoe commented:
So I'm a -1 for pretty printing in log files because we should be sending this to stuff like ELK and those log lines get parsed and split up with the leading context. As soon as you force newlines in that messes that up.

stevekeay commented:
> So I'm a -1 for pretty printing in log files because we should be sending this to stuff like ELK and those log lines get parsed and split up with the leading context. As soon as you force newlines in that messes that up.

Fair enough, I removed the pretty-printing and rebased on main.
